README.md

```diff
@@ -49,28 +49,37 @@ Behind the scenes the following occurs in both cases:
 * Publishes all this information.
 
 
-## Dropwizard
+## Metrics
 
-Out of the box this uses [Dropwizard metrics](https://dropwizard.github.io/metrics/3.1.0/) to report the following stats.
-The following metrics are reported which have the form `scientist.[experiment name].*`:
+Out of the box this uses a `NoopMetricsProvider`. This is not very useful. You’ll want to pick use of the built-in providers or implement your own `MetricsProvider`, which will report to your chosen registry:
```

Comment:
I'm confused why we would not just make `Dropwizard` the default and allow people to change to `Micrometer` or their own custom metrics provider? It seems this will just cause confusion for new developers. I'm curious your motivation for this default.

Reply:
I think I see why now as this allows the `Dropwizard` dependency to be optional. I'm a bit worried that many people will just skip this section of the README and end up using the useless `NoopMetricsProvider` and have a bad experience with the library.  I wonder if there is anything else we can add that would reduce this potential issue. Perhaps the `NoopMetricsProvider` at the very least should log warnings to users that they should specify a metrics provider?

Reply:
If Dropwizard is the default, that will require Dropwizard metrics to be present on the class path, since `Experiment` will statically create a `DatadogMetricsProvider` (1) and its default `MetricRegistry` (2):

1. https://github.com/rawls238/Scientist4J/blob/dbb41cda74bb1ff857e307189f6e58572669e21b/Scientist4JCore/src/main/java/com/github/rawls238/scientist4j/Experiment.java#L22
2. https://github.com/rawls238/Scientist4J/blob/3de7c69d676dc165f620e1438b4565e472e50d15/Scientist4JCore/src/main/java/com/github/rawls238/scientist4j/metrics/DropwizardMetricsProvider.java#L13

The rationale for the no-op default was to change less of the API, but thinking more about this, I agree that the no-op default is pretty confusing—everyone would be supplying a provider of their choice anyway.

Reply:
@rawls238 since this is a breaking change anyway, what do you think about updating the `Experiment` constructor to require a `MetricsProvider`?

An alternative would be to do some a reflection or `ClassLoader`-based check for whether Datadog or Micrometer were present, but that still complicates things for a user who wants neither of those providers, and creates something totally custom.

Reply:
yes I like that alternative much more. Let's do that

Reply:
What about having it actually fail with no provider specified?  This would cause a fast fail that would help developers discover the error sooner rather than later.

Reply:
@rawls238 sorry, just making sure—you like updating the constructor or check the class loader?

Reply:
Updating constructor or failing if no provider is specified are both reasonable to me. Agree that the class loader seems too complicated

Reply:
Done in 8b9af20